packages/sit-onyx/src/styles/mixins/text.scss

```
@@ -0,0 +1,8 @@
+/**
+* Function for getting a calculated CSS "ch" unit value that can be used to e.g. set the width of an element based on the text length.
+* Includes a bug fix for webkit/safari so the text is not truncated.
+*/
+@function ch($length) {
+  // the 1px offset is needed to fix a bug in webkit/safari where the text is truncated
+  @return calc($length * 1ch + var(--onyx-1px-in-rem));
+}
```

Imho this is not a bug but a quirk of the definition of the `ch` unit: It's an size approximation based on the `0` character of the current font, but browsers can always decide to just use `0.5em` instead. So for (not mono-)fonts it will always be kind of inaccurate.

```suggestion
/**
* Function for getting a approximated CSS "ch" length value that can be used to e.g. set the width of an element based on the text length.
* Adds 1px of additional space to ensure no overflow for low character counts.
*/
@function ch($length) {
  @return calc($length * 1ch + var(--onyx-1px-in-rem));
}
```
(For an exact solution we would need to use the trick that we use with the textarea.)